src/compose/service-manager.coffee

```diff
@@ -0,0 +1,276 @@
+Promise = require 'bluebird'
+_ = require 'lodash'
+EventEmitter = require 'events'
+JSONParser = require 'jsonparse'
+fs = Promise.promisifyAll(require('fs'))
+
+logTypes = require '../lib/log-types'
+{ checkInt } = require '../lib/validation'
+constants = require '../lib/constants'
+
+Service = require './service'
+#restartVars = (conf) ->
+#	return _.pick(conf, [ 'RESIN_DEVICE_RESTART', 'RESIN_RESTART' ])
+
+module.exports = class ServiceManager extends EventEmitter
+	constructor: ({ @docker, @logger, @config }) ->
+		@containerHasDied = {}
+		@listening = false
+		# Volatile state of containers, indexed by containerId (or random strings if we don't have a containerId yet)
+		@volatileState = {}
+
+	killAllLegacy: =>
+		# Containers haven't been normalized (this is an updated supervisor)
+		# so we need to stop and remove them
+		@docker.getImage(constants.supervisorImage).inspect()
+		.then (supervisorImage) =>
+			Promise.map(@docker.listContainers(all: true), (container) =>
+				@docker.getImage(container.Image).inspect()
```

Comment:
Docker 1.8 and above includes `ImageID` in the container list response. If we don't support earlier versions you can remove the inspect

Reply:
Good one, done